src/printf.hpp

```diff
@@ -25,7 +26,9 @@ struct printf_descriptor {
 };
 
 using printf_descriptor_map_t = std::unordered_map<uint32_t, printf_descriptor>;
+typedef std::function<void(const char*, size_t)> printf_callback_func;
```

Comment:
This is not the prototype defined by the `cl_arm_printf` [extension](https://registry.khronos.org/OpenCL/extensions/arm/cl_arm_printf.html#_modifications_to_the_opencl_api_specification):
```
(void)(*callback)(const char *buffer, size_t len, size_t complete, void *user_data)
```